chef/cookbooks/cpe_screensaver/resources/cpe_screensaver.rb

```diff
@@ -39,30 +46,178 @@
+    'PayloadUUID' => '3B2AD6A9-F99E-4813-980A-4147617B2E75',
+    'PayloadEnabled' => true,
+    'PayloadDisplayName' => 'ScreenSaver',
+    'PayloadContent' => {
+      'com.apple.screensaver' => {
+        'Forced' => [
+          {
+            'mcx_preference_settings' => {
+              'idleTime' => node['cpe_screensaver']['idleTime'],
+              'askForPassword' =>
+                node['cpe_screensaver']['askForPassword'],
+              'askForPasswordDelay' =>
+                node['cpe_screensaver']['askForPasswordDelay'],
+            },
+          },
+        ],
+      },
+    },
+  )
+
+  if node['cpe_screensaver']['MESSAGE']
+    module_name = 'Computer Name'
+    message = node['cpe_screensaver']['MESSAGE']
+    screensaver_profile['PayloadContent'].push(
+      'PayloadType' => 'com.apple.ManagedClient.preferences',
+      'PayloadVersion' => 1,
+      'PayloadIdentifier' => "#{prefix}.screensaver.Basic",
+      'PayloadUUID' => '8dd9a983-f59b-47c3-a408-ac55d287cbd4',
+      'PayloadEnabled' => true,
+      'PayloadDisplayName' => message,
+      'PayloadContent' => {
+        'com.apple.screensaver.Basic.ByHost' => {
+          'Set-Once' => [
+            {
+              'mcx_preference_settings' => {
+                'MESSAGE' => message,
+              },
+            },
+          ],
+        },
+      },
+    )
+  end
+
+  if node['cpe_screensaver']['MESSAGE'] ||
+     node['cpe_screensaver']['SelectedFolderPath']
+    resources = '/System/Library/Frameworks/ScreenSaver.framework/Resources/'
+    module_name = module_name ? module_name : 'iLifeSlideshows'
+    path = resources + module_name + '.saver'
+
+    if node['cpe_screensaver']['SelectedFolderPath'].to_s.include? '.saver'
+      resources = '/Library/Screen Savers/'
+      module_name = node['cpe_screensaver']['SelectedFolderPath']
+      path = resources + module_name
+    end
+
+    screensaver_profile['PayloadContent'].push(
+      'PayloadType' => 'com.apple.ManagedClient.preferences',
+      'PayloadVersion' => 1,
+      'PayloadIdentifier' => "#{prefix}.screensaver.ByHost",
+      'PayloadUUID' => '01cb34f8-36f8-4fb6-babc-5ae3aa6c165b',
+      'PayloadEnabled' => true,
+      'PayloadDisplayName' => module_name,
+      'PayloadContent' => {
+        'com.apple.screensaver.ByHost' => {
+          'Set-Once' => [
+            {
+              'mcx_preference_settings' => {
+                'moduleDict' => {
                   'idleTime' => node['cpe_screensaver']['idleTime'],
-                  'askForPassword' =>
-                    node['cpe_screensaver']['askForPassword'],
-                  'askForPasswordDelay' =>
-                    node['cpe_screensaver']['askForPasswordDelay'],
+                  'moduleName' => module_name,
+                  'path' => path,
+                  'type' => 0,
                 },
               },
-            ],
-          },
+            },
+          ],
         },
       },
-    ],
-  }
+    )
+  else
```

Comment:
so we have this thing about elses.... If the else is smaller(less lines) than the if condition then the 'if' logic should be inverted. The thought is, people will lose the else condition when reading the code.